src/transformers/models/dac/modeling_dac.py

```diff
@@ -0,0 +1,653 @@
+# limitations under the License.
+"""Transformers DAC model."""
+
+import math
+from dataclasses import dataclass
+from typing import Optional
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.utils import weight_norm
+
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    ModelOutput,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from .configuration_dac import DacConfig
+
+
+# General docstring
+_CONFIG_FOR_DOC = "DacConfig"
+
+
+@dataclass
+class DacOutput(ModelOutput):
+    """
+    Args:
+        loss (`torch.Tensor`):
+            Loss from the encoder model, comprising the weighted combination of the commitment and codebook losses.
+        audio_values (`torch.Tensor` of shape `(batch_size, 1, input_length)`):
+            Decoded audio data.
+        quantized_representation (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`):
+            Quantized continuous representation of input.
+        codebook_indices (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`):
+            Codebook indices for each codebook (quantized discrete representation of input).
+        projected_latents (`torch.Tensor` of shape `(batch_size, num_codebooks * dimension, time_steps)`):
+            Projected latents (continuous representation of input before quantization).
+    """
+
+    loss: torch.FloatTensor = None
+    audio_values: torch.FloatTensor = None
+    quantized_representation: torch.FloatTensor = None
+    codebook_indices: torch.FloatTensor = None
+    projected_latents: torch.FloatTensor = None
+
+
+@dataclass
+class DacEncoderOutput(ModelOutput):
+    """
+    Args:
+        loss (`torch.Tensor`):
+            Loss from the encoder model, comprising the weighted combination of the commitment and codebook losses.
+        quantized_representation (`torch.Tensor` of shape `(batch_size, dimension, time_steps)`, *optional*):
+            Quantized continuous representation of input.
+        codebook_indices (`torch.Tensor` of shape `(batch_size, num_codebooks, time_steps)`, *optional*):
+            Codebook indices for each codebook (quantized discrete representation of input).
+        projected_latents (`torch.Tensor` of shape `(batch_size, num_codebooks * dimension, time_steps)`, *optional*):
+            Projected latents (continuous representation of input before quantization).
+    """
+
+    loss: torch.FloatTensor = None
+    quantized_representation: torch.FloatTensor = None
+    codebook_indices: torch.FloatTensor = None
+    projected_latents: torch.FloatTensor = None
+
+
+@dataclass
+class DacDecoderOutput(ModelOutput):
+    """
+    Args:
+        audio_values (`torch.Tensor` of shape `(batch_size, 1, input_length)`):
+            Decoded audio data.
+    """
+
+    audio_values: torch.FloatTensor = None
```

Comment:
I see that it's similar to `EncodecDecoderOutput` but with very few differences. In general, note that in transformers, we seek to minimize new pieces of code whenever it's possible.

This lead me to two comments / questions:
1. why do we return a 3D tensor instead of a 2D tensor of shape `(batch_size, input_length)` ?
2. If you finally decide to go for a 2D tensor, you could simply use a Copied from statement like this one: 

For example, here, you could add:

```suggestion
@dataclass
# Copied from transformers.models.encodec.modeling_encodec.EncodecDecoderOutput with Encodec->Dac, segment_length->input_length
class DacDecoderOutput(ModelOutput):
    """
    Args:
        audio_values (`torch.FloatTensor`  of shape `(batch_size, input_length)`, *optional*):
            Decoded audio values, obtained using the decoder part of Dac.
    """

    audio_values: torch.FloatTensor = None
```

Reply:
We can definitely return a 2D tensor! we returned a 3D tensor mostly to mimic the original model behaviour.